### PR TITLE
Post controller

### PR DESCRIPTION
**Pull request by anaghayawale**

### CODE REVIEW

File: src/controllers/post.controller.js [hunk 1]
```diff
@@ -24,58 +25,65 @@ const addNewPost = asyncHandler(async (req, res, next) => {
...
 
-// ------------------------- Update Post -------------------------
+// ------------------------- Update self Post -------------------------
 const updatePost = asyncHandler(async (req, res, next) => {
-  const { postId, image, caption, likes, comments } = req.body;
+  const { postId, caption } = req.body;
+  const userIdFromToken = req.user._id;
 
-  if (!(caption && postId)) {
+  if (bodyDataExists(postId && caption)) {
```
Comment (sunilband) [inline src/controllers/post.controller.js:33]: is this the condition that you are going for ?

Comment (anaghayawale) [reply src/controllers/post.controller.js:33]: yes

File: src/controllers/post.controller.js [hunk 2]
```diff
@@ -24,58 +25,65 @@ const addNewPost = asyncHandler(async (req, res, next) => {
   res.status(201).json(new ApiResponse(201, { newPost }, "Post created"));
 });
 
-// ------------------------- Update Post -------------------------
+// ------------------------- Update self Post -------------------------
 const updatePost = asyncHandler(async (req, res, next) => {
-  const { postId, image, caption, likes, comments } = req.body;
+  const { postId, caption } = req.body;
```
Comment (sunilband) [inline src/controllers/post.controller.js:30]: What if user does not want to update caption but just image ?

Comment (anaghayawale) [reply src/controllers/post.controller.js:30]: user cannot update image. i dont want to add such feature

File: src/controllers/post.controller.js [hunk 3]
```diff
@@ -24,58 +25,65 @@ const addNewPost = asyncHandler(async (req, res, next) => {
...
   }
-  if (likes) {
-    if (likes.length === 0) throw new ApiError(400, "Incomplete Data");
+
+  if (existingPost.userId.toString() != userIdFromToken) {
+    throw new ApiError(403, "Unauthorized");
   }
 
-  const existingPost = await Post.findOneAndUpdate(
+  const updatedPost = await Post.findOneAndUpdate(
```
Comment (sunilband) [inline src/controllers/post.controller.js]: You already fetched the post, just update that instead of doing one more DB call

Comment (anaghayawale) [reply src/controllers/post.controller.js]: ok.

File: src/controllers/post.controller.js [hunk 4]
```diff
@@ -24,58 +25,65 @@ const addNewPost = asyncHandler(async (req, res, next) => {
...
 const deletePost = asyncHandler(async (req, res, next) => {
   const { postId } = req.body;
+  const userIdFromToken = req.user._id;
 
   if (!postId) {
     throw new ApiError(400, "Incomplete Data");
   }
 
-  const existingPost = await Post.findOneAndDelete({ _id: postId });
+  const existingPost = await Post.findOne({ _id: postId });
```
Comment (sunilband) [inline src/controllers/post.controller.js:69]: for delete no need to check existing

File: src/routes/post.route.js [hunk 5]
```diff
@@ -11,8 +11,8 @@ const router = Router();
 
 // POST
```
Comment (sunilband) [inline src/routes/post.route.js]: protected route comment

**Review by sunilband — CHANGES_REQUESTED** [review]
check the comments

**Pull request merged**